### DISCUSSION TITLE
How to cast local files?

### DISCUSSION THREAD

**Discussion by abdemirza** [react-native-google-cast/react-native-google-cast#390] [Q&A]
I am working on a project where I am required to cast local files from android to chromecast devices but google cast only supports url, can anyone please help me out with this one ?

**Comment by MattWeiler** [comment]
I'm looking to do this as well.

The way that I see it, there are 2 options:


# Base64 Encoded URLs

For this option, you can just `Base64` encode your files, and then build a URL with them.

Example:
```
data:audio/mpeg;base64,UklGRoAqAABXQVZ...2324234==
```

The major downside of this approach is that when I tried casting audio files this say, that were approximately 20 seconds long, they all failed.
No errors were thrown, they just didn't work.
But when I tried with a very short audio file, approximately 1.5 seconds long, it worked fine.

I guess there are limitations on the length of the URL.


# Locally Hosted HTTP Server

I found the below post that details the steps of how to use the `@dr.pogodin/react-native-static-server` library with `React Native Expo`:
https://stackoverflow.com/a/76219615/1827349

I haven't had the time to try this yet, but it does look promising.
